src/gui/entry/EditEntryWidget.cpp

```diff
@@ -446,6 +446,12 @@ void EditEntryWidget::saveEntry()
 
 void EditEntryWidget::acceptEntry()
 {
+    // Check if passwords are mismatched first to prevent saving
+    if (!passwordsEqual()) {
+        showMessage(tr("Different passwords supplied."), MessageWidget::Error);
+        return;
+    }
```

Comment:
@droidmonkey do you think you can add a test to `TestGui::testEditEntry()` for that?

Reply:
Good idea, on it

Reply:
@louib ready for you